pyro/distributions/transforms/spline.py

```diff
@@ -8,18 +8,22 @@
 # * https://github.com/tonyduan/normalizing-flows/blob/master/nf/flows.py; and,
 # * https://github.com/hmdolatabadi/LRS_NF/blob/master/nde/transforms/nonlinearities.py,
 # under the MIT license.
+
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
-from torch.distributions import constraints
+from torch.distributions import Transform, constraints
 
+from pyro.distributions.conditional import ConditionalTransformModule
 from pyro.distributions.torch_transform import TransformModule
 from pyro.distributions.util import copy_docs_from
+from pyro.nn import DenseNN
 
 
 def _search_sorted(bin_locations, inputs, eps=1e-6):
     """
-    Searches for which bin an input belongs to (in a way that is parallelizable and amenable to autodiff)
+    Searches for which bin an input belongs to (in a way that is parallelizable and
```

Comment:
FYI it looks like `torch.searchsorted()` is coming soon https://github.com/pytorch/pytorch/pull/34577 https://github.com/pytorch/pytorch/issues/7284 I guess we should try to follow that interface to make it easy to switch once it is released.

Reply:
I've changed the signature of `_searchsorted` so it matches `torch.searchsorted()`. The latter has many more features, however, you should be able to replace `_searchsorted` with `torch.searchsorted` as is without breaking anything